### PR TITLE
[A/B Test 3] Integrate A/B test service with package search

### PR DESCRIPTION
**Pull request by joelverhagen**
Depends on https://github.com/NuGet/NuGetGallery/pull/7311
Address https://github.com/NuGet/NuGetGallery/issues/7166

This toggles the search client used for package search based on the user's cookie. This also toggles the telemetry emitted.

### CODE REVIEW

File: tests/NuGetGallery.Facts/Controllers/PackagesControllerFacts.cs [hunk 1]
```diff
@@ -317,6 +327,30 @@ private static Mock<IPackageUploadService> GetValidPackageUploadService(string p
...
+            {
+                var type = typeof(PackagesController);
+
+                var constructors = type.GetConstructors();
+                var constructor = Assert.Single(constructors);
+
+                var parameters = constructor.GetParameters();
+                var parameter = Assert.Single(
+                    parameters,
+                    p => p.Name == DefaultDependenciesModule.ParameterNames.PackagesController_PreviewSearchService);
```
Comment (loic-sharma) [inline tests/NuGetGallery.Facts/Controllers/PackagesControllerFacts.cs]: Nice 👍

File: src/NuGetGallery/Controllers/PackagesController.cs [hunk 2]
```diff
@@ -120,6 +122,7 @@ public partial class PackagesController
             IUserService userService,
             IMessageService messageService,
             ISearchService searchService,
+            ISearchService previewSearchService,
```
Comment (shishirx34) [inline src/NuGetGallery/Controllers/PackagesController.cs]: may be add a comment to not accidentally modify this name since its keyed to the DI?

Comment (shishirx34) [reply src/NuGetGallery/Controllers/PackagesController.cs]: I see the test, nice.

Comment (joelverhagen) [reply src/NuGetGallery/Controllers/PackagesController.cs]: Yeah I will leave the comment out since there is a test.

**Pull request merged**